src/main/java/com/umc/networkingService/domain/board/service/BoardServiceImpl.java

```diff
@@ -0,0 +1,232 @@
+                request.getTitle(),
+                request.getContent(),
+                semesterPermission);
+
+        boardFileService.updateBoardFiles(board, files);
+
+        return new BoardIdResponse(board.getId());
+    }
+
+    @Override
+    @Transactional
+    public BoardIdResponse deleteBoard(Member member, UUID boardId) {
+        Board board = loadEntity(boardId);
+
+        boardFileService.deleteBoardFiles(board);
+        board.delete();
+
+        return new BoardIdResponse(board.getId());
+
+    }
+
+
+    public List<Semester> checkPermission(Member member, HostType hostType, BoardType boardType) {
+        //현재 기수
+        Semester nowSemester = Semester.findActiveSemester();
+
+        //허용 기수 (기본 모든 기수에게 허용)
+        List<Semester> permissionSemesters = Arrays.stream(Semester.values()).toList();
+
+        //boardType에 따라 권한 Check
+        switch (boardType) {
+            case OB -> checkPermissionForOBBoard(member, nowSemester);
+            case NOTICE -> checkPermissionForNoticeBoard(member, hostType);
+            case WORKBOOK -> {
+                checkPermissionForWorkbookBoard(member, hostType);
+                //워크북의 경우 현재 기수만 볼 수 있도록 허용
+                permissionSemesters = List.of(nowSemester);
+            }
+        }
+        return permissionSemesters;
+    }
+
+
+    //OB 게시판 권한 확인 함수
+    public void checkPermissionForOBBoard(Member member, Semester nowSemester) {
+        // OB -> Semester의 isActive가 활성화되지 않은 사용자만 가능
+        if (member.getRecentSemester().equals(nowSemester))
+            throw new RestApiException(ErrorCode.FORBIDDEN_MEMBER);
+    }
+
+    //공지사항 게시판 권한 확인 함수
+    public void checkPermissionForNoticeBoard(Member member, HostType hostType) {
+        //멤버의 역할
+        Role memberRole = member.getRole();
+        //Center 권한이 있는 멤버의 역할
+        List<Role> centerStaffRoles = Role.centerStaffRoles();
+        //Branch 권한이 있는 멤버의 역할
+        Role branchStaffRole = Role.branchStaffRole();
+        //Campus 권한이 있는 멤버의 역할
+        List<Role> campusStaffRoles = Role.campusStaffRoles();
+
+        // 공지사항 권한 CHECK
+        if (hostType == HostType.CENTER) {
+            // CENTER && NOTICE -> ROLE이 TOTAL_STAFF OR CENTER_STAFF 인 사람
+            if (!centerStaffRoles.contains(memberRole)) {
+                throw new RestApiException(ErrorCode.FORBIDDEN_MEMBER);
+            }
+        } else if (hostType == HostType.BRANCH) {
+            // BRANCH && NOTICE -> ROLE이 BRANCH_STAFF인 사람
+            if (branchStaffRole != memberRole) {
+                throw new RestApiException(ErrorCode.FORBIDDEN_MEMBER);
+            }
+            // BRANCH 나머지 -> 해당 지부 사람들 모두
+        } else if (hostType == HostType.CAMPUS) {
+            // CAMPUS && NOTICE -> ROLE이 CAMPUS_STAFF, STAFF인 사람
+            if (!campusStaffRoles.contains(memberRole)) {
+                throw new RestApiException(ErrorCode.FORBIDDEN_MEMBER);
+            }
+            // CAMPUS 나머지 -> 해당 campus 사람들 모두
```

Comment:
HostType 열거형 클래스에 Role 열거형 클래스처럼 priority 필드를 주어서 비교하면 좋을 거 같아요!!!
아래는 CENTER = 3, BRANCH = 4, CAMPUS = 5의 priority를 지정해주었을 때 예시입니다!!!
```suggestion
        if (member.getRole().getPriority() >= hostType.getPriority())
            throw new RestApiException(ErrorCode.FORBIDDEN_MEMBER);
```

Reply:
WOW....갓벡스 🥹🥹🥹🥹🥹🥹🥹 감사합니다..